core/worker.go

```diff
@@ -498,34 +502,12 @@ func (w *worker) asyncStateLoop() {
 						}).Fatal("Go-Quai Panicked")
 					}
 				}()
-				if side.ResetUncles {
-					w.uncleMu.Lock()
-					w.localUncles = make(map[common.Hash]*types.WorkObjectHeader)
-					w.remoteUncles = make(map[common.Hash]*types.WorkObjectHeader)
-					w.uncleMu.Unlock()
-				}
 				for _, wo := range side.Blocks {
-
 					// Short circuit for duplicate side blocks
-					w.uncleMu.RLock()
-					if _, exists := w.localUncles[wo.Hash()]; exists {
-						w.uncleMu.RUnlock()
+					if _, exists := w.Uncles.Get(wo.Hash()); exists {
```

Comment:
(nitpick) you could use Contains here, grabs the RLock instead of Lock which might be slightly faster

Reply:
Cool, changed, thanks